dev-python/shiboken/shiboken-5.12.3.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2019 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} )
+CMAKE_MIN_VERSION=3.1.0
+
+inherit cmake-utils llvm python-r1
+
+MY_P="pyside-setup-everywhere-src-${PV}"
+
+DESCRIPTION="Python binding generator for C++ libraries"
+HOMEPAGE="https://wiki.qt.io/PySide2"
+SRC_URI="https://download.qt.io/official_releases/QtForPython/pyside2/PySide2-${PV}-src/${MY_P}.tar.xz"
+
+# The "sources/shiboken2/libshiboken" directory is triple-licensed under the
+# GPL v2, v3+, and LGPL v3. All remaining files are licensed under the GPL v3
+# with version 1.0 of a Qt-specific exception enabling shiboken2 output to be
+# arbitrarily relicensed. (TODO)
+LICENSE="|| ( GPL-2 GPL-3+ LGPL-3 ) GPL-3"
+SLOT="2"
+KEYWORDS="~amd64 ~x86"
+IUSE="+docstring numpy test"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+BDEPEND=">=sys-devel/clang-6.0.0:="
+RDEPEND="${PYTHON_DEPS}
+	>=dev-qt/qtcore-${PV}:5
+	docstring? (
+		>=dev-libs/libxml2-2.6.32
+		>=dev-libs/libxslt-1.1.19
+		>=dev-qt/qtxml-${PV}:5
+		>=dev-qt/qtxmlpatterns-${PV}:5
+	)
+	numpy? ( dev-python/numpy[${PYTHON_USEDEP}] )
+"
+DEPEND="${RDEPEND}
+	test? ( >=dev-qt/qttest-${PV}:5 )
+"
+
+S="${WORKDIR}/${MY_P}/sources/shiboken2"
+DOCS=( AUTHORS )
+
+# Ensure the path returned by get_llvm_prefix() contains clang as well.
+llvm_check_deps() {
+	has_version "sys-devel/clang:${LLVM_SLOT}"
+}
+
+src_prepare() {
+	#FIXME: File an upstream issue requesting a sane way to disable NumPy support.
+	if ! use numpy; then
+		sed -i -e '/print(os\.path\.realpath(numpy))/d' \
+			libshiboken/CMakeLists.txt || die
+	fi
+
+	# CMakeLists.txt assumes clang builtin includes are installed under
+	# LLVM_INSTALL_DIR. They are not on Gentoo. See bug 624682.
+	sed -i -e "s~clangPathLibDir = findClangLibDir()~clangPathLibDir = QLatin1String(\"${EPREFIX}/usr/lib\")~" \
+		ApiExtractor/clangparser/compilersupport.cpp || die
+
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	shiboken_configure() {
+		local mycmakeargs=(
+			-DBUILD_TESTS=$(usex test)
+			-DDISABLE_DOCSTRINGS=$(usex !docstring)
+			-DPYTHON_CONFIG_SUFFIX="-${EPYTHON}"
+			-DPYTHON_EXECUTABLE="${PYTHON}"
+			-DPYTHON_SITE_PACKAGES="$(python_get_sitedir)"
```

Comment:
**I wish.** Like any dying coral reef ecosystem, `sources/shiboken2/CMakeLists.txt` slowly accreted over the course of several years by several different engineers who loathed each other as well as comments and documentation. Notably:

* `PYTHON_CONFIG_SUFFIX` defaults to insane values for Python 3 and must thus be manually overwritten. (*Sucks, I know. Took me forever to get this one right, too.*)
* `PYTHON_EXECUTABLE` derives from the prior ebuild. I could never actually figure out how the makefile deduced this string. From what I can tell, it's never actually set anywhere. I'm usually wrong, though. I don't know `cmake` as well as I'd like. Perhaps the `find_package(PythonInterp ${ARGV0} REQUIRED)` line in `data/shiboken_helpers.cmake` magically sets this and other Python-centric variables?
* `PYTHON_SITE_PACKAGES` derives from the prior ebuild, too. I did figure out how the makefile deduced this string – but the conditional logic looked a bit dodgy, so I preserved this as is.

You're right, though. Let's try omitting `PYTHON_EXECUTABLE` and `PYTHON_SITE_PACKAGES`.

Reply:
last time I worked on this package (years ago, I know...) specifying `PYTHON_EXECUTABLE` was the only way to convince cmake to use a specific python version (instead of whatever `/usr/bin/python` points to), but upstream has finally added `USE_PYTHON_VERSION` which *may* have made the former obsolete.

I don't remember why we were also specifying `PYTHON_SITE_PACKAGES` though...